From 7c9abd06f57275d109d6b22a577c7c80dd007543 Mon Sep 17 00:00:00 2001
From: jammyjellyfish <jammyjellyfish255@outlook.com>
Date: Wed, 19 Apr 2023 15:55:06 +0800
Subject: [PATCH] Fix prototype warnings

The flag -Wdeprecated-non-prototype is enabled by
default in Clang 15. This patch try to fix compile
error when building with Clang 15(and gcc with
-Wstrict-prototypes flag.)
---
 src/libopensc/card-openpgp.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/libopensc/card-openpgp.c b/src/libopensc/card-openpgp.c
index fad32f0..ea1d874 100644
--- a/src/libopensc/card-openpgp.c
+++ b/src/libopensc/card-openpgp.c
@@ -129,7 +129,7 @@ static pgp_ec_curves_t	ec_curves_gnuk[] = {
 
 static int		pgp_get_card_features(sc_card_t *card);
 static int		pgp_finish(sc_card_t *card);
-static void		pgp_iterate_blobs(pgp_blob_t *, void (*func)());
+static void		pgp_iterate_blobs(pgp_blob_t *blob, void (*func)(pgp_blob_t *));
 
 static int		pgp_get_blob(sc_card_t *card, pgp_blob_t *blob,
 				 unsigned int id, pgp_blob_t **ret);
@@ -1150,7 +1150,7 @@ pgp_free_blob(pgp_blob_t *blob)
  * Internal: iterate through the blob tree, calling a function for each blob.
  */
 static void
-pgp_iterate_blobs(pgp_blob_t *blob, void (*func)())
+pgp_iterate_blobs(pgp_blob_t *blob, void (*func)(pgp_blob_t *))
 {
 	if (blob) {
 		pgp_blob_t *child = blob->files;
-- 
2.40.0

